pyccel/codegen/printing/cwrappercode.py

```diff
@@ -519,6 +646,14 @@ def get_PyBuildValue(self, used_names, variable):
     #                 _print_ClassName functions
     #--------------------------------------------------------------------
 
+    def _print_VariableAddress(self, expr):
```

Comment:
is it necessary to overite this method? why can't we use `CCodePrinter`'s method for that?

Reply:
I agree. Fixed: 18470e4

Reply:
Hmm, it seems this is more complicated than expected as `stored_in_c_pointer` is not used as described everywhere. Especially when ndarrays are involved. I am trying to unravel the mess

Reply:
I notice that the `CCodePrinter`'s method doesn't add an `&` if `expr.variable.rank > 0`, while the one you wrote in `cwrappercode` does.

Reply:
Yes exactly. This is really just a sloppy workaround though, it would be better to stick to the definition where `stored_in_c_pointer` does what it says and `VariableAddress` returns the C address of a variable. Currently that is the case for 0d objects but nd objects are all treated differently. This is fine as long as an `ndarray` cannot be a pointer, but optional ndarray arguments are pointers so this now needs handling properly

Reply:
exactly, We need to be careful about that.

Reply:
`VariableAddress` now does what it is supposed to